src/libraries/Pool.sol

```diff
@@ -307,7 +310,16 @@ library Pool {
 
         bool exactInput = params.amountSpecified < 0;
 
-        if (params.amountSpecified == 0) return (BalanceDeltaLibrary.ZERO_DELTA, 0, swapFee, state);
+        // a swap fee of MAX_FEE_PIPS (100%) makes exact output swaps impossible since the input is entirely consumed by the fee
+        if (swapFee == LPFeeLibrary.MAX_LP_FEE) {
+            if (!exactInput) {
+                InvalidFeeForExactOut.selector.revertWith();
+            }
+        }
+
+        // 0 is the fee amount that should be paid to the protocol
+        // swapFee is the pool's fee in pips (LP fee + protocol fee)
```

Comment:
I think the first line could be something more around motivation for no protocol fee in this specific case like:

// when the amount swapped is 0, there is no protocolFee applied and the fee amount paid to protocol is set to 0